src/storage/UpdateVertexProcessor.cpp

```diff
@@ -0,0 +1,364 @@
+    if (retNum > 0) {
+        nebula::cpp2::Schema respScheam;
+        respScheam.columns.resize(retNum);
+        for (auto& tc : this->tagContexts_) {
+            for (auto& prop : tc.props_) {
+                if (prop.returned_) {
+                    auto column = this->columnDef(std::move(prop.prop_.name), prop.type_.type);
+                    respScheam.columns[prop.retIndex_] = std::move(column);
+                }
+            }
+        }
+        resp_.set_schema(std::move(respScheam));
+        resp_.set_return_data(std::move(respData_));
+    }
+}
+
+
+kvstore::ResultCode UpdateVertexProcessor::collectVertexProps(
+                            PartitionID partId,
+                            VertexID vId,
+                            TagID tagId,
+                            const std::vector<PropContext>& props,
+                            FilterContext* fcontext,
+                            Collector* collector) {
+    auto prefix = NebulaKeyUtils::prefix(partId, vId, tagId);
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = this->kvstore_->prefix(spaceId_, partId, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        VLOG(3) << "Error! ret = " << static_cast<int32_t>(ret) << ", spaceId " << spaceId_;
+        return ret;
+    }
+    // Will decode the properties according to the schema version
+    // stored along with the properties
+    tagReader_[tagId] = std::make_unique<KeyReaderPair>();
+    if (iter && iter->valid()) {
+        tagReader_[tagId]->key = iter->key().toString();
+        tagReader_[tagId]->reader = RowReader::getTagPropReader(this->schemaMan_,
+                                                                iter->val(),
+                                                                spaceId_,
+                                                                tagId);
+    } else if (insertable_ && updateTagIDs_.find(tagId) != updateTagIDs_.end()) {
+        tagReader_[tagId]->insert = true;
+        auto now = std::numeric_limits<int64_t>::max() - time::WallClock::fastNowInMicroSec();
+        tagReader_[tagId]->key = NebulaKeyUtils::vertexKey(partId, vId, tagId, now);
+        auto schema = this->schemaMan_->getTagSchema(spaceId_, tagId);
+        RowWriter writer(schema);
+        tagReader_[tagId]->reader = RowReader::getRowReader(writer.encode(), schema);
+    } else {
+        VLOG(3) << "Missed partId " << partId << ", vId " << vId << ", tagId " << tagId;
+        return kvstore::ResultCode::ERR_KEY_NOT_FOUND;
+    }
+    this->collectProps(tagReader_[tagId]->reader.get(),
+                       tagReader_[tagId]->key,
+                       props,
+                       fcontext,
+                       collector);
+    return ret;
+}
+
+
+kvstore::ResultCode UpdateVertexProcessor::processVertex(PartitionID partId,
+                                                         VertexID vId) {
+    tagReader_.clear();
```

Comment:
this function is too long, maybe should split it.

Reply:
OK, recheck it now please!